rest_framework/routers.py

```diff
@@ -219,7 +219,7 @@ def get_lookup_regex(self, viewset, lookup_prefix=''):
         # consume `.json` style suffixes and should break at '/' boundaries.
         lookup_field = getattr(viewset, 'lookup_field', 'pk')
         lookup_url_kwarg = getattr(viewset, 'lookup_url_kwarg', None) or lookup_field
-        lookup_value = getattr(viewset, 'lookup_value_regex', '[^/.]+')
```

Comment:
I guess the `[^/.]+` was originally there in order to support our suffix handling for eg. `.json` so this change might have some unintended side-effects?

Reply:
With the default of having a trailing slash it shouldn't make any difference. When using `.json` in one of the encapsulating regexes it should still match both: https://www.debuggex.com/r/xz5zlKxfItbuC0Hh

Reply:
Alternatively the match could be made non-greedy: `[^/]+?`